### PR TITLE
[react-navigation]: added types in navigate params

### PR DESCRIPTION
**Pull request by ArmandoAssuncao**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by timwangdev** [comment]
Flow type for `NavigationScreenProp` in official project does not include params generic, which is the most of this definition referenced.

https://github.com/react-navigation/react-navigation/blob/master/flow/react-navigation.js#L473-L499

What's the opinions of other maintainers? Should we keep sync with the flow type?

File: types/react-navigation/index.d.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@
 //                 Veit Lehmann: <https://github.com/levito>
 //                 Roberto Huertas: <https://github.com/robertohuertasm>
 //                 Steven Miller <https://github.com/YourGamesBeOver>
+//                 Armando Assuncao <https://github.com/ArmandoAssuncao>
```
Comment (iRoachie) [inline types/react-navigation/index.d.ts]: Adding your name here means you'll be one of the maintainers for these typings, and you'll get notifications for issues and PRs to this definition. Just checking if you understand what it means

**Review by iRoachie — CHANGES_REQUESTED** [review]
Confirm becoming a maintainer

**Comment by ArmandoAssuncao** [comment]
@iRoachie yep, i understood about becoming maintainer.

**Review by iRoachie — APPROVED** [review]

**Pull request merged**